doc/timeline.md

```diff
@@ -0,0 +1,205 @@
+
+struct MyComponentWithKeyFrame : public Ra::Engine::Component
+{
+   MyComponentWithKeyFrame( const std::string& name, Ra::Engine::Entity* entity, Scalar value = 0_ra )
+       : Ra::Engine::Component( name, entity )
+       , m_keyframes( value, 0_ra )
+       , m_currentValue( value )
+   {
+       // use the linear interpolator instead of the default one
+       m_keyframes.setInterpolator( Ra::Core::Animation::interpolate<Scalar> );
+   }
+
+   void initialize() override {}
+   void goTo( Scalar t ) {
+       // Here we use the value at time t.
+       m_currentValue = m_keyframes.at( t );
+   }
+   void update() {
+       // Here we can update other data that depend on the value.
+   }
+
+   KeyFrames m_keyframes;
+   Scalar m_currentValue;
+};
+
+struct MyTimeDependantSystem : public Ra::Engine::AbstractTimedSystem
+{
+   MyTimeDependantSystem( Ra::GuiBase::Timeline* timeline )
+       : Ra::Engine::AbstractTimedSystem()
+       , m_timeline( timeline )
+   {}
+
+   void generateTasks( Ra::Core::TaskQueue* taskQueue, const Ra::Engine::FrameInfo& ) override {
+       // Update all Components data.
+       for ( auto compEntry : m_components )
+       {
+           auto comp = static_cast<MyComponentWithKeyFrame*>( compEntry.second );
+           auto func = std::bind( &MyComponentWithKeyFrame::update, comp );
+           auto task = new Ra::Core::FunctionTask( func, "MyUpdateTask" );
+           taskQueue->registerTask( task );
+       }
+   }
+   void AnimationSystem::handleAssetLoading( Ra::Engine::Entity* entity, const Ra::Core::Asset::FileData* fileData ) {
+       // Add a Component whenever a file is loaded
+       auto comp = new MyComponentWithKeyFrame( "MyComponentWithKeyFrame_", entity );
+       registerComponent( entity, comp );
+       if ( m_timeline )
+       {
+           // on insertion we may want to get the frame value from elsewhere
+           auto inserter = [comp]( const Scalar& t ) {
+               // for example value(t) + 1
+               comp->m_keyframes.insertKeyFrame( t, comp->m_keyframes.at( t ) + 1_ra );
+           };
+           // No need for an update callback since the component owns the KeyFramedValue.
+           m_timeline->registerKeyFramedValue( comp, "KeyFrame", &comp->m_keyframes, inserter );
+       }
+   }
+   void goTo( Scalar t ) override {
+       // Update all Components' KeyFramedValue.
+       for ( auto compEntry : m_components )
+       {
+           compEntry.second->goTo( t );
+       }
+   }
+   void cacheFrame( const std::string& dir, uint frameID ) const override {}
+   bool restoreFrame( const std::string& dir, uint frameID ) override {}
+
+   Ra::GuiBase::Timeline* m_timeline{nullptr};
+};
+```
+
+## Dynamic KeyFramedValues:
+Those are `Ra::Core::Animation::KeyFramedValue`s that are not part of an `Ra::Engine::Entity`, `Ra::Engine::Component` or
+`Ra::Engine::RenderObject` data, but are used to keyframe some of its data.
```

Comment:
I don't understand the nuance between _is not part_ and _used to keyframe some of its data_. Would holds it in that second case ? This should be clear at this stage of the doc (I see I can find some answers in the example above), otherwise it is very difficult to follow.

Reply:
`Part of` means the Component has its data already keyframed through one of its attributes, like for the example above.
`used to keyframe some of its data` means it is not the case initially, but the data can still be made keyframed externally (e.g. by an other system/plugin). In this case, the KeyFramedValue attached to this data is owned by the UI and directly updates the data.